source/fe/fe_values.cc

```diff
@@ -5029,8 +5029,18 @@ FESubfaceValues<dim, spacedim>::reinit(
 {
   Assert(face_no < GeometryInfo<dim>::faces_per_cell,
          ExcIndexRange(face_no, 0, GeometryInfo<dim>::faces_per_cell));
-  Assert(subface_no < cell->face(face_no)->n_children(),
+  // We would like to check for subface_no < cell->face(face_no)->n_children(),
+  // but unfortunately the current function is also called for
+  // faces without children (see
```

Comment:
This is not specifically related to the matrix-free code, but the general case when one wants to do periodicity with hanging nodes - in case one side of a periodic boundary is more refined, the coarse side wants to reinit onto a subface, but it does not have children itself. So I suggest to update the comment regarding this more general statement in terms of periodicity. Probably, you should even be able to check for the case of a periodic face in the assertion here.

Reply:
This is true (but method is only used within matrixfree). Should I create a separate PR?

Reply:
No, it is fine to do it here. Just try to update the assertion to that case.